File: src/features/typing/data/go/longest_palindromic_substring.go
package main

func longestPalindrome(s string) string {
	start, maxLen := 0, 0
	expand := func(left, right int) {
		for left >= 0 && right < len(s) && s[left] == s[right] {
			left--
			right++
		}
		if right-left-1 > maxLen {
			start = left + 1
			maxLen = right - left - 1
		}
	}
	for i := 0; i < len(s); i++ {
		expand(i, i)
		expand(i, i+1)
	}
	return s[start : start+maxLen]
}

File: src/features/typing/data/ts/longest_palindromic_substring.ts
function longestPalindrome(s: string): string {
  let start = 0;
  let maxLen = 0;
  const expand = (l: number, r: number): void => {
    while (l >= 0 && r < s.length && s[l] === s[r]) {
      l--;
      r++;
    }
    const len = r - l - 1;
    if (len > maxLen) {
      start = l + 1;
      maxLen = len;
    }
  };
  for (let i = 0; i < s.length; i++) {
    expand(i, i);
    expand(i, i + 1);
  }
  return s.slice(start, start + maxLen);
}
